### PR TITLE
Fine grained control over SSH requests

### PR DESCRIPTION
**Pull request by 0xfede7c8**

### CODE REVIEW

File: default_options.h [hunk 1]
```diff
@@ -46,9 +46,6 @@ IMPORTANT: Some options will require "make clean" after changes */
  * much traffic. */
 #define DROPBEAR_SMALL_CODE 1
 
-/* Enable X11 Forwarding - server only */
-#define DROPBEAR_X11FWD 1
```
Comment (dgutson) [inline default_options.h:50]: please restore this and control it from local_options.

Comment (dgutson) [inline default_options.h:50]: please restore this and control it from local_options.

File: default_options.h [hunk 2]
```diff
@@ -59,7 +56,6 @@ IMPORTANT: Some options will require "make clean" after changes */
 #define DROPBEAR_SVR_REMOTETCPFWD 1
 
 /* Enable Authentication Agent Forwarding */
-#define DROPBEAR_SVR_AGENTFWD 1
```
Comment (dgutson) [inline default_options.h:62]: ditto

File: svr-chansession.c [hunk 3]
```diff
@@ -367,30 +367,52 @@ static void chansessionrequest(struct Channel *channel) {
...
-	if (strcmp(type, "window-change") == 0) {
+#if DROPBEAR_WINDOW_CHANGE_REQUEST
+	if (strcmp(type, "window-change") == 0)
+	{
 		ret = sessionwinchange(chansess);
 	} 
-#if DROPBEAR_SHELL
-	else if (strcmp(type, "shell") == 0) {
+#else
+	if(0) {}
```
Comment (dgutson) [inline svr-chansession.c]: move this to top

File: svr-chansession.c [hunk 4]
```diff
@@ -367,30 +367,52 @@ static void chansessionrequest(struct Channel *channel) {
...
 #endif
 #if DROPBEAR_SVR_AGENTFWD
-	} else if (strcmp(type, "auth-agent-req@openssh.com") == 0) {
+	else if (strcmp(type, "auth-agent-req@openssh.com") == 0)
+	{
 		ret = svr_agentreq(chansess);
+	}
 #endif
-	} else if (strcmp(type, "signal") == 0) {
+	else if (strcmp(type, "signal") == 0)
```
Comment (dgutson) [inline svr-chansession.c]: what about this one?

**Review by dgutson — CHANGES_REQUESTED** [review]

File: default_options.h [hunk 5]
```diff
@@ -292,10 +288,18 @@ be overridden at runtime with -I. 0 disables idle timeouts */
 /* The default path. This will often get replaced by the shell */
 #define DEFAULT_PATH "/usr/bin:/bin"
 
-/* Enable the terminal execution from the client. */
-#define DROPBEAR_TERMINAL 1
```
Comment (dgutson) [inline default_options.h:296]: ditto

File: default_options.h (hunk 1 above)
Comment (0xfede7c8) [reply default_options.h:50]: ? Check bottom

Comment (0xfede7c8) [reply default_options.h:50]: ditto

File: default_options.h (hunk 2 above)
Comment (0xfede7c8) [reply default_options.h:62]: ditto

File: svr-chansession.c (hunk 3 above)
Comment (0xfede7c8) [reply svr-chansession.c]: Done.

File: svr-chansession.c (hunk 4 above)
Comment (0xfede7c8) [reply svr-chansession.c]: Done.

File: svr-chansession.c [hunk 6]
```diff
@@ -367,13 +367,12 @@ static void chansessionrequest(struct Channel *channel) {
 	chansess = (struct ChanSess*)channel->typedata;
 	dropbear_assert(chansess != NULL);
 	TRACE(("type is %s", type))
+	if(0) {}
 #if DROPBEAR_WINDOW_CHANGE_REQUEST
 	if (strcmp(type, "window-change") == 0)
```
Comment (dgutson) [inline svr-chansession.c]: else if

**Review by dgutson — APPROVED** [review]
Also consider please INET_D / NO_INET_D

File: default_options.h (hunk 5 above)
Comment (0xfede7c8) [reply default_options.h:296]: unused

File: svr-chansession.c (hunk 6 above)
Comment (0xfede7c8) [reply svr-chansession.c]: done

**Pull request merged**